src/main/java/uk/gov/hmcts/cmc/claimstore/documents/content/FullAdmissionResponseContentProvider.java

```diff
@@ -0,0 +1,49 @@
+package uk.gov.hmcts.cmc.claimstore.documents.content;
+
+import com.google.common.collect.ImmutableMap;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.cmc.domain.models.PaymentOption;
+import uk.gov.hmcts.cmc.domain.models.RepaymentPlan;
+import uk.gov.hmcts.cmc.domain.models.response.FullAdmissionResponse;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+import static uk.gov.hmcts.cmc.claimstore.services.staff.content.RepaymentPlanContentProvider.create;
+
+@Component
+public class FullAdmissionResponseContentProvider {
+
+    private final StatementOfMeansContentProvider statementOfMeansContentProvider;
+
+    public FullAdmissionResponseContentProvider(
+        StatementOfMeansContentProvider statementOfMeansContentProvider
+    ) {
+        this.statementOfMeansContentProvider = statementOfMeansContentProvider;
+    }
+
+    public Map<String, Object> createContent(FullAdmissionResponse fullAdmissionResponse) {
+        requireNonNull(fullAdmissionResponse);
+
+        PaymentOption type = fullAdmissionResponse.getPaymentOption();
+
+        ImmutableMap.Builder<String, Object> contentBuilder = new ImmutableMap.Builder<String, Object>()
+            .put("responseTypeSelected", fullAdmissionResponse.getResponseType().getDescription())
+            .put("paymentOption", type.getDescription());
+
+        Optional<RepaymentPlan> optionalRepaymentPlan = fullAdmissionResponse.getRepaymentPlan();
```

Comment:
I do not think we have to say optional in variable name since the type is `Optional<RepaymentPlan>`. A `if (repaymentPlan.isPresent()) {` would read well enough for me.

Reply:
refactored and using ifPresent closure.